handler/implementations/maxIntBaseHandler.go

```diff
@@ -256,6 +271,9 @@ func (h *MaxIntBaseHandler) pushFile(
 	// agents that write to the same sysctl. That's because there is no guarantee
 	// that the other host agent will read-after-write and retry as sysbox does.
 
+	h.Lock.Lock()
```

Comment:
I'm a bit concerned that we will eventually introduce a bug in this code that will end up in deadlock. Can you add a note somewhere (not sure where's the proper location: handler struct definition or container one), to define the lock-acquisition protocol that we'll follow? container-ext-lock -> container-int-lock -> handler-lock.

Reply:
Good idea, I added the text in the description of the handlerBase object.

// HandlerBase is a type common to all handlers                                                                                                                                                                                                                                                                               
//                                                                                                                                                                                                                                                                                                                            
// Note: the "Lock" variable can be used to synchronize across concurrent                                                                                                                                                                                                                                                     
// executions of the same handler (e.g., if multiple processes within the same                                                                                                                                                                                                                                                
// sys container or across different sys containers are accessing the same                                                                                                                                                                                                                                                    
// sysbox-fs emulated resource). When obtaining the handler lock, only do so to                                                                                                                                                                                                                                               
// synchronize accesses to host resources associated with the emulated resources                                                                                                                                                                                                                                              
// (e.g., if a handler needs to write to the host's procfs for example). While                                                                                                                                                                                                                                                
// holding the handler lock, avoid accessing objects of "container" struct type as                                                                                                                                                                                                                                            
// those have a dedicated lock which is typically held across invocations of the                                                                                                                                                                                                                                              
// handler lock. Violating this rule may result in deadlocks.